ClassPg.playground/Contents.swift

```diff
@@ -0,0 +1,266 @@
+        if move(distance: distance) {
+            print("I can transport many people! And \(passengersCount) passengers too!")
+        }
+    }
+}
+
+
+class Truck: Car {
+
+    func transportCargo(cargoWeight: Double, distance: Double) {
+        if move(distance: distance + cargoWeight / 5) {
+            print("I'm strong! I'm transport cargo weighing \(cargoWeight) tons!")
+        }
+    }
+}
+
+
+enum CarType: String {
+    case sedan
+    case offroad
+    case bus
+    case truck
+}
+
+
+class Factory {
+
+    private var _carCounter: Int
+    private var _name: String
+    private var _sedanParking: [Sedan]
+    private var _offroadParking: [Offroad]
+    private var _busParking: [Bus]
+    private var _truckParking: [Truck]
+    
+    var carCounter: Int {
+        get {
+            return _carCounter
+        }
+    }
+    var name: String {
+        get {
+            return _name
+        }
+    }
+
+    init(name: String) {
+        _name = name
+        _carCounter = 0
+        _sedanParking = []
+        _offroadParking = []
+        _busParking = []
+        _truckParking = []
+    }
+
+    private func createSedan(engine: Engine, driveUnit: DriveUnit, maxFuel: Double) -> Void {
+        _sedanParking.append(
+            Sedan(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel, manufactorer: self)
+        )
+    }
+
+    private func createOffroad(engine: Engine, driveUnit: DriveUnit, maxFuel: Double) -> Void {
+        _offroadParking.append(
+            Offroad(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel, manufactorer: self)
+        )
+    }
+
+    private func createBus(engine: Engine, driveUnit: DriveUnit, maxFuel: Double) -> Void {
+        _busParking.append(
+            Bus(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel, manufactorer: self)
+        )
+    }
+
+    private func createTruck(engine: Engine, driveUnit: DriveUnit, maxFuel: Double) -> Void {
+        _truckParking.append(
+            Truck(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel, manufactorer: self)
+        )
+    }
+
+    func releaseCars(carType: CarType, engine: Engine, driveUnit: DriveUnit, maxFuel: Double, carsCount: Int) -> Void {
+        for _ in 0..<carsCount {
+            _carCounter += 1
+
+            switch carType {
+            case .sedan:
+                createSedan(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel)
+            case .offroad:
+                createOffroad(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel)
+            case .bus:
+                createBus(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel)
+            case .truck:
+                createTruck(engine: engine, driveUnit: driveUnit, maxFuel: maxFuel)
+            }
+        }
+    }
+
+    func getReleasedSedans() -> [Sedan] {
+        var result: [Sedan] = []
+
+        for _ in 0..<_sedanParking.count {
+            result.append(_sedanParking.removeLast())
+        }
+
+        return result
```

Comment:
Зачем если можно сразу возвращать созданные автомобили
```suggestion
       return sedanParking
```